File: src/main/java/net/atos/zac/app/zaken/model/RESTZaakBetrokkeneGegevens.java
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2023-2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.app.zaken.model;

import java.util.UUID;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import net.atos.zac.app.klanten.model.klant.IdentificatieType;

public class RESTZaakBetrokkeneGegevens {

    @NotNull public UUID zaakUUID;

    @NotNull public UUID roltypeUUID;

    @NotBlank public String roltoelichting;

    @NotNull public IdentificatieType betrokkeneIdentificatieType;

    @NotBlank public String betrokkeneIdentificatie;
}

File: src/main/kotlin/net/atos/zac/app/zaken/model/RESTZaakBetrokkeneGegevens.kt
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.app.zaken.model

import jakarta.validation.constraints.NotBlank
import net.atos.zac.app.klanten.model.klant.IdentificatieType
import nl.lifely.zac.util.AllOpen
import nl.lifely.zac.util.NoArgConstructor
import java.util.UUID

@AllOpen
@NoArgConstructor
data class RESTZaakBetrokkeneGegevens(
    var zaakUUID: UUID,

    var roltypeUUID: UUID,

    @field:NotBlank
    var roltoelichting: String,

    var betrokkeneIdentificatieType: IdentificatieType,

    @field:NotBlank
    var betrokkeneIdentificatie: String
)
